evmcore/tx_validation_test.go

```diff
@@ -421,26 +389,62 @@ func TestValidateTx_Signer_RejectsTxWith(t *testing.T) {
 				types.NewPragueSigner(big.NewInt(2)), key)
 			require.NoError(t, err)
 
-			// validate transaction
-			err = validateTx(signedTx, getTestTransactionsOption())
+			_, err = ValidateTxForNetworkRules(signedTx, getTestNetworkRules())
 			require.ErrorIs(t, err, ErrInvalidSender)
 		})
 	}
 }
 
-func TestValidateTx_Balance_RejectsTxWhen(t *testing.T) {
+func TestValidateTxForNetworkRules_AcceptsTxWith(t *testing.T) {
+	for name, tx := range getTxsOfAllTypes() {
+		t.Run(name, func(t *testing.T) {
+			opt := getTestNetworkRules()
+			overBaseFee := new(big.Int).Add(opt.currentBaseFee, big.NewInt(1))
+			underMaxGas := opt.currentMaxGas - 1
+
+			setGasPriceOrFeeCap(t, tx, overBaseFee)
+			setGas(t, tx, underMaxGas)
+
+			_, signedTx := signTxForTest(t, tx)
+			_, err := ValidateTxForNetworkRules(signedTx, opt)
+			require.NoError(t, err)
+		})
+	}
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// State Validation
+
+func TestValidateTxForState_Nonce_RejectsTxWith(t *testing.T) {
+	for name, tx := range getTxsOfAllTypes() {
+		t.Run(fmt.Sprintf("older nonce/%v", name), func(t *testing.T) {
+
+			// set nonce lower than the current account nonce
+			currentNonce := uint64(2)
```

Comment:
nit: should be taken from the `tx` instead of hard-coding it

Reply:
https://github.com/0xsoniclabs/sonic/pull/200/commits/730cc402e58a98f26bdde28704fbde8a849ccae9